src/screens/ScanQrCodeScreen.tsx

```diff
@@ -0,0 +1,169 @@
+import { useEffect, useState } from 'react';
+import {
+  Animated,
+  View,
+  Text,
+  Pressable,
+  Button,
+  StyleSheet,
+} from 'react-native';
+import { IconButton } from 'react-native-paper';
+import { BarCodeScanner } from 'expo-barcode-scanner';
+// import { getDemoCred } from '../credentials';
+// import { getDemoRel, getUserId } from '../relationships';
+// import {
+//   getDid,
+//   importContact,
+//   importVerifiedCredential,
+// } from '../roots';
+import React from 'react';
+import { CompositeScreenProps } from '@react-navigation/core/src/types';
+import { BarCodeEvent } from 'expo-barcode-scanner/src/BarCodeScanner';
+import { styles } from '../styles/styles';
+import { ConfigService } from '../services';
+import { useDispatch, useSelector } from 'react-redux';
+import { addCredential } from '../store/slices/credential';
+import { getVerifiedCredentials } from '../store/selectors/credential';
+
+const configService = new ConfigService();
+
+export default function ScanQrCodeScreen({
+  route,
+  navigation,
+}: CompositeScreenProps<any, any>) {
+  console.log('Scan QR - route params', route.params);
+  const [hasPermission, setHasPermission] = useState<boolean>(false);
+  const [scanned, setScanned] = useState<boolean>(false);
+  const [timeOutId, setTimeOutId] = useState<NodeJS.Timeout>();
+  const verifiedCredentials = useSelector(getVerifiedCredentials);
+  const dispatch = useDispatch();
+  const modelType = route.params.type;
+
+  const addDummyCredenial = () => {
+    dispatch(
+      addCredential({
+        alias: `dummy${verifiedCredentials.length + 1}_credentialAlias`,
+        verifiedCredential: {
+          encodedSignedCredential: `dummy${verifiedCredentials.length + 1}_verifiedCredential_${Date.now()}`,
+          proof: {
+            hash: `dummy${verifiedCredentials.length + 1}_proofHash_${Date.now()}`,
+            index: verifiedCredentials.length,
+          },
+        },
+      })
+    );
+  }
+
+  const handleDemo = async () => {
+    if (!scanned && configService.getDemo()) {
+      setScanned(true);
+      console.log('Scan QR - pretending to scan with demo data');
+      alert('No data scanned, using demo data instead.');
+      
+      if (modelType === 'contact') {
```

Comment:
When you hit the scan button from the contact list, the dummy data should be a contact that is added to the contact list.  Similar to https://github.com/roots-id/rootswallet/blob/main/src/screens/ScanQRCodeScreen.tsx#L36

Reply:
Also, I see errors when I scan real QR codes.  On scan can we post a console message saying scan complete, but only using mock data?
` WARN  Possible Unhandled Promise Rejection (id: 1):
SyntaxError: JSON Parse error: Unexpected identifier "https"
parse@[native code]
http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:182911:34
generatorResume@[native code]
asyncGeneratorStep@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:25017:26
_next@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:25039:29
http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:25046:14
tryCallTwo@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:30558:9
doResolve@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:30722:25
Promise@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:30581:14
http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:25035:25
handleBarCodeScanned@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:182924:27
http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:183086:21
invokeGuardedCallbackProd@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:3109:21
invokeGuardedCallback@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:3213:42
invokeGuardedCallbackAndCatchFirstError@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:3217:36
executeDispatch@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:3293:48
executeDispatchesInOrder@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:3313:26
executeDispatchesAndRelease@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:4662:35
forEachAccumulated@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:3845:18
runEventsInBatch@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:4686:27
runExtractedPluginEventsInBatch@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:4766:25
http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:4741:42
batchedUpdates$1@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:17151:20
batchedUpdates@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:4648:36
_receiveRootNodeIDEvent@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:4740:23
receiveEvent@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:4770:32
__callFunction@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:23976:36
http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:23704:31
__guard@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:23930:15
callFunctionReturnFlushedQueue@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:23703:21
callFunctionReturnFlushedQueue@[native code]
`

Reply:
weird I did not notice that. Will take a look